src/GraphQL/Extensions/SchemaExtensions.cs

```diff
@@ -405,6 +410,102 @@ private void Replace(IProvideResolvedType provider)
+        var linkImportGraphType = new LinkImportGraphType() { Name = config.NameForType("Import") };
+        var linkDirective = new Directive("link")
+        {
+            Arguments = new QueryArguments(
+                new QueryArgument(typeof(NonNullGraphType<StringGraphType>)) { Name = "url" },
+                new QueryArgument(typeof(StringGraphType)) { Name = "as" },
+                new QueryArgument(linkPurposeGraphType) { Name = "purpose" },
+                new QueryArgument(new ListGraphType(linkImportGraphType)) { Name = "import" }),
+            Repeatable = true,
+        };
+        linkDirective.Locations.Add(DirectiveLocation.Schema);
+        schema.Directives.Register(linkDirective);
+    }
+
+    /// <summary>
+    /// Adds a @link directive to the schema, specifying a URL and optional configuration for the link.
+    /// If the url specified has already been linked, you may apply additional changes to the link.
+    /// </summary>
+    /// <param name="schema">The schema to which the directive will be added.</param>
+    /// <param name="url">The URL of the linked schema.</param>
+    /// <param name="configuration">An optional action to configure the link further.</param>
+    public static void LinkSchema(this ISchema schema, string url, Action<LinkConfiguration>? configuration = null)
+    {
+        var linkInstalled = false;
+        var appliedDirectives = schema.GetAppliedDirectives();
+        if (appliedDirectives != null)
+        {
+            foreach (var appliedDirective in appliedDirectives)
+            {
+                if (appliedDirective.Name == "link")
+                {
+                    var urlMatch = appliedDirective.FindArgument("url")?.Value as string;
+                    if (urlMatch == url)
+                    {
+                        // parse existing configuration
+                        if (!LinkConfiguration.TryParseDirective(appliedDirective, true, out var link))
+                            throw new InvalidOperationException("Unable to parse existing @link directive for this url.");
+                        // add additional configuration
+                        configuration?.Invoke(link);
+                        // re-apply directive
+                        link.ConfigureAppliedDirective(appliedDirective);
+                        return;
+                    }
+                    if (url == LinkConfiguration.LINK_URL)
+                        linkInstalled = true;
+                }
+            }
+        }
+        if (!linkInstalled && url != LinkConfiguration.LINK_URL)
+        {
+            schema.AddLinkDirectiveSupport();
+        }
+
+        var config = new LinkConfiguration(url);
+        configuration?.Invoke(config);
+        schema.ApplyDirective("link", config.ConfigureAppliedDirective);
+    }
+
+    /// <summary>
+    /// Gets the applied @link directives from the schema.
+    /// Results are only valid after the schema has been initialized.
```

Comment:
Should throw then if `schema.Initialized == false`?

Reply:
I added checks